src/common/utils/convertCpu.ts

```diff
@@ -3,18 +3,38 @@
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
 
+import { TypedRegEx } from "typed-regex";
+
 // Helper to convert CPU K8S units to numbers
 
-const thousand = 1000;
-const million = thousand * thousand;
-const shortBillion = thousand * million;
+const unitConverters = new Map([
+  ["n", 1000 ** -3],
+  ["u", 1000 ** -2],
+  ["m", 1000 ** -1], // milli
+  ["", 1000 ** 0], // no units
+  ["k", 1000 ** 1],
+  ["M", 1000 ** 2],
+  ["G", 1000 ** 3],
+  ["P", 1000 ** 4],
+  ["T", 1000 ** 5],
+  ["E", 1000 ** 6],
+]);
+
+const cpuUnitsRegex = TypedRegEx("^(?<digits>\\d+)(?<unit>.*)$");
+
+export function cpuUnitsToNumber(value: string) {
+  const match = cpuUnitsRegex.captures(value);
+
+  if (!match) {
+    return undefined;
+  }
 
-export function cpuUnitsToNumber(cpu: string) {
-  const cpuNum = parseInt(cpu);
+  const { digits, unit } = match;
+  const conversion = unitConverters.get(unit);
 
-  if (cpu.includes("m")) return cpuNum / thousand;
-  if (cpu.includes("u")) return cpuNum / million;
-  if (cpu.includes("n")) return cpuNum / shortBillion;
+  if (conversion === undefined) {
+    return undefined;
+  }
 
-  return parseFloat(cpu);
+  return parseInt(digits) * conversion;
```

Comment:
Oh, couldn't the number be a float? You can have 0.5 for cpu, no? https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu:~:text=Fractional%20requests%20are%20allowed

Reply:
Maybe, but having tried it kubernetes seems to convert a YML file with `0.5` for the CPU into `500m`

Reply:
Yes parsing float would probably be better. Will change.